### PR TITLE
Enable building on Mac M1

### PR DESCRIPTION
**Pull request by jamesrowe08**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,7 +52,7 @@ $(BLS512_LIB): $(OBJ_DIR)/bls_c512.o
 $(BLS384_256_LIB): $(OBJ_DIR)/bls_c384_256.o
 	$(AR) $@ $<
 
-ifneq ($(findstring $(OS),mac/mingw64),)
+ifneq ($(findstring $(OS),mac/mac-m1/mingw64),)
```
Comment (gupadhyaya) [inline Makefile:55]: will this not break the non-m1 mac case?

File: Makefile [hunk 2]
```diff
@@ -97,15 +97,15 @@ $(EXE_DIR)/%256_test.exe: $(OBJ_DIR)/%256_test.o $(BLS256_LIB) $(MCL_LIB)
 # sample exe links libbls256.a
 $(EXE_DIR)/%.exe: $(OBJ_DIR)/%.o $(BLS256_LIB) $(MCL_LIB)
 	$(PRE)$(CXX) $< -o $@ $(BLS256_LIB) -L$(MCL_DIR)/lib -lmcl $(LDFLAGS)
-ifeq ($(OS),mac)
+ifneq ($(findstring $(OS),mac/mac-m1),)
```
Comment (gupadhyaya) [inline Makefile]: same here. might break the non-m1 Mac case.

**Review by gupadhyaya — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (jamesrowe08) [reply Makefile]: This was a typo and fixed here.

File: Makefile (hunk 1 above)
Comment (jamesrowe08) [reply Makefile:55]: This is including mac-m1 with the normal mac path

Comment (jamesrowe08) [reply Makefile:55]: It should still work like usual with non m1

File: Makefile [hunk 3]
```diff
@@ -97,15 +97,15 @@ $(EXE_DIR)/%256_test.exe: $(OBJ_DIR)/%256_test.o $(BLS256_LIB) $(MCL_LIB)
...
+ifeq ($(findstring $(OS),mac/mac-m1),)
 	install_name_tool bin/bls_smpl.exe -change lib/libmcl.dylib $(MCL_DIR)/lib/libmcl.dylib
 endif
 
 SAMPLE_EXE=$(addprefix $(EXE_DIR)/,$(SAMPLE_SRC:.cpp=.exe))
 sample: $(SAMPLE_EXE)
 
 TEST_EXE=$(addprefix $(EXE_DIR)/,$(TEST_SRC:.cpp=.exe))
-ifeq ($(OS),mac)
+ifneq ($(findstring $(OS),mac/mac-m1),)
```
Comment (gupadhyaya) [inline Makefile:108]: this also seems flipped?

**Review by gupadhyaya — APPROVED** [review]
LGTM. thanks.

**Pull request merged**